services/httppost/service.go

```diff
@@ -27,25 +27,55 @@ func NewService(c Configs, l *log.Logger) *Service {
 	return s
 }
 
+// Only one of name and url should be non-empty
+type Endpoint struct {
+	s      *Service
+	name   string
+	url    string
```

Comment:
As a preference I would like to see Endpoint as having a url field and not know its name. Then for the Endpoints created with just a URL use the URL as the name of the endpoint in the map. 

This way all endpoints are the same and just reference a URL and a set of headers.

Also, where are the headers? Are they in the config struct still?